src/Uno.UWP/Storage/Streams/DataWriter.cs

```diff
@@ -0,0 +1,394 @@
+			byte[] guidBytes = value.ToByteArray();
+			WriteInt32(BitConverter.ToInt32(guidBytes, 0));
+			WriteInt16(BitConverter.ToInt16(guidBytes, 4));
+			WriteInt16(BitConverter.ToInt16(guidBytes, 6));
+			WriteInt64(BitConverter.ToInt64(guidBytes, 8));
+		}
+
+		private void WriteBytes(byte[] bytes, int size)
+		{
+			// maybe we should reverse array (if platform endianess is different than requested)
+			if ((ByteOrder == ByteOrder.LittleEndian && !BitConverter.IsLittleEndian)
+				|| (ByteOrder == ByteOrder.BigEndian && BitConverter.IsLittleEndian))
+			{
+				Array.Reverse(bytes);
+			}
+
+			_memStream.Write(bytes, 0, size);
+		}
+
+		public void WriteInt16(short value) => WriteBytes(BitConverter.GetBytes(value), 2);
+		public void WriteInt32(int value) => WriteBytes(BitConverter.GetBytes(value), 4);
+		public void WriteInt64(long value) => WriteBytes(BitConverter.GetBytes(value), 8);
+		public void WriteUInt16(ushort value) => WriteBytes(BitConverter.GetBytes(value), 2);
+		public void WriteUInt32(uint value) => WriteBytes(BitConverter.GetBytes(value), 4);
+		public void WriteUInt64(ulong value) => WriteBytes(BitConverter.GetBytes(value), 8);
+		public void WriteSingle(float value)
+		{   // yes, it is endianness dependant
+			WriteBytes(BitConverter.GetBytes(value), 4);
+		}
+		public void WriteDouble(double value)
+		{   // yes, it is endianness dependant
+			WriteBytes(BitConverter.GetBytes(value), 8);
+		}
+		public void WriteDateTime(DateTimeOffset value)
+		{ // it seems like documentation error...
+		  // implementation here is based on testing how it works on UWP
+			var epoch1600 = new DateTime(1601, 1, 1, 0, 0, 0);
+			long ticks = (value - epoch1600.ToLocalTime()).Ticks;
+			WriteInt64(ticks);
+		}
+		public void WriteTimeSpan(TimeSpan value) => WriteInt64(value.Ticks);
+
+		public uint MeasureString(string value)
+		{
+			switch (this.UnicodeEncoding)
+			{
+				case UnicodeEncoding.Utf16BE:
+					return (uint)(Encoding.Unicode.GetBytes(value).Length);
+				case UnicodeEncoding.Utf16LE:
+					return (uint)(Encoding.BigEndianUnicode.GetBytes(value).Length);
+				default: // UnicodeEncoding.Utf8
+					return (uint)(Encoding.UTF8.GetBytes(value).Length);
+			}
+		}
+
+		public uint WriteString(string value)
+		{
+			byte[] dump;
+
+			switch (this.UnicodeEncoding)
+			{
+				case UnicodeEncoding.Utf16BE:
+					dump = Encoding.Unicode.GetBytes(value);
+					break;
+				case UnicodeEncoding.Utf16LE:
+					dump = Encoding.BigEndianUnicode.GetBytes(value);
+					break;
+				default: // UnicodeEncoding.Utf8
+					dump = Encoding.UTF8.GetBytes(value);
+					break;
+			}
+			WriteBytes(dump);
+			return (uint)dump.Length;
+		}
+
+		public void WriteBuffer(IBuffer buffer)
+		{
+			WriteBytes(buffer.Data);
+		}
+
+		public void WriteBuffer(IBuffer buffer, uint start, uint count)
+		{
+			byte[] partBuffer = new byte[count];
+			for (int i = 0; i < count; i++)
+			{
+				partBuffer[i] = buffer.Data[start + i];
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'buffer' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305560)